lib/aur-chroot--standalone

```diff
@@ -0,0 +1,346 @@
+
+# Print paths to container and used makepkg/pacman paths. This does
+# not require a priorly created container.
+if (( status )); then
+    printf 'chroot:%s\npacman:%s\nmakepkg:%s\n' "$directory" "$pacman_conf" "$makepkg_conf"
+    exit 0
+fi
+
+is_multilib=0
+cache_dirs=()
+while read -r key _ value; do
+    case $key=$value in
+        Server=file://*)
+            # bind mount file:// paths to container (#461)
+            # required for update/build steps
+            bindmounts_rw+=("${value#file://}") ;;
+        CacheDir=*)
+            cache_dirs+=("${value}")
+            bindmounts_rw+=("${value}") ;;
+        \[multilib\]=)
+            is_multilib=1 ;;
+    esac
+done < <(pacman-conf --config "$pacman_conf")
+wait "$!"
+
+cleanup() {
+    rm "$directory"/root/aur-chroot--helper
+}
+
+trap cleanup EXIT
+
+# create new container, required for update/build steps
+if [[ ! -d $directory/root ]]; then
+    if (( !create )); then
+        printf >&2 '%s: %s is not a directory\n' "$argv0" "$directory"/root
+        printf >&2 '%s: did you run aur chroot --create?\n' "$argv0"
+        exit 20
+    fi
+
+    # default to base-devel or multilib-devel, unless packages are
+    # specified on the command-line.
+    # available packages depend on the configured pacman configuration
+    if (( $# )); then
+        base_packages=("$@")
+    elif (( is_multilib )) && [[ $machine == "x86_64" ]]; then
+        base_packages=('base-devel' 'multilib-devel')
+    else
+        base_packages=('base-devel')
+    fi
+
+    # # create both the root container and a directory for pacman db
+    install -d "$directory"/root/var/lib/pacman -m 755
+
+    # # XXX: allow setting machine-id from host (and provide default) to enable .nspawn file usage
+    systemd-firstboot --root="$directory"/root --copy-locale --copy-timezone --setup-machine-id
+
+    pacman -Sy "${base_packages[@]}" \
+        --config "$pacman_conf" \
+        -r "$directory"/root \
+        "${cache_dirs[@]/#/--cachedir=}" \
+        --noconfirm
+
+    {
+        declare -f _chroot_create
+        printf '_chroot_create || exit\n'
+    } >> "$directory"/root/aur-chroot--helper
+fi >&2
+
+if (( update )); then
+    printf 'pacman -Syu --noconfirm --config %q %s || exit' "$pacman_conf" "${*@Q}" >> "$directory"/root/aur-chroot--helper
+fi >&2
+
+if (( create )) || (( update )); then
+    systemd-nspawn -D "$directory"/root "${bindmounts_ro[@]/#/--bind-ro=}" "${bindmounts_rw[@]/#/--bind=}" /bin/bash /aur-chroot--helper
+
+    exit
+fi >&2
+
+if (( build )); then
+    # add configurable directories for makepkg (and PWD as BUILDDIR)
+    bindmounts_rw+=(
+        "$PWD"
+        "${PKGDEST:-$PWD}:/pkgdest"
+        "${LOGDEST:-$PWD}:/logdest"
+        "${SRCDEST:-$PWD}:/srcdest"
+        "${SRCPKGDEST:-$PWD}:/srcpkgdest"
+    )
+
+    # create makepkg_user
+    systemd-sysusers --root "$directory"/root --replace=/usr/lib/sysusers.d/build.conf - \
```

Comment:
Maybe it makes sense to use a second helper for the `--build` step, or overwrite (`>`) the contents. As it stands, the commands added for the `--create` and `--update` steps would be run a second time when both `--build` and `--create` / `--update` are specified.